File: lab7/task2/cpp/ValuableJewelry.cpp
#include <iostream>
#include "Jewelry.cpp"
#ifndef VALUABLEJEWELRY_H
#define VALUABLEJEWELRY_H
using namespace std;

class ValuableJewelry : public Jewelry {
  public:
    double getFullPricePerGramm();

    ValuableJewelry();
    ValuableJewelry(double arg_weight, double arg_pricePerGramm);
};

double ValuableJewelry::getFullPricePerGramm() {
  return Jewelry::getFullPricePerGramm() * 2;
}

ValuableJewelry::ValuableJewelry() : Jewelry() {}

ValuableJewelry::ValuableJewelry(double arg_weight, double arg_pricePerGramm) : Jewelry(arg_weight, arg_pricePerGramm) {}

#endif

File: lab7/task2/cs/task2/ValuableJewelry.cs
namespace task2 {
    public class ValuableJewelry : Jewelry {
        
        public double GetFullPricePerGramm() {
          return base.GetFullPricePerGramm() * 2;
        }
        
        public ValuableJewelry() : base() {
        }

        public ValuableJewelry(double arg_weight, double arg_pricePerGramm) : base(arg_weight, arg_pricePerGramm) {
        }
    }
}